src/realm/dictionary.cpp

```diff
@@ -549,6 +541,7 @@ std::pair<Dictionary::Iterator, bool> Dictionary::insert(Mixed key, Mixed value)
         throw StaleAccessor("Stale dictionary");
     }
 
+    bool add_salt = value.is_type(type_Dictionary, type_List);
```

Comment:
nit: don't really like the variable name. maybe `insert_collection` is better?

Reply:
Changed